File: Practice/programmers_038.py
def solution(n):
    answer = ''
    while n != 0:
        remain = n % 3
        if remain == 0:
            answer = '4' + answer
            n = n // 3 - 1
        else:
            answer = str(remain) + answer
            n = n // 3
    return answer

# 문자열을 이용한 풀이
def solution(n):
    num = ['1','2','4']
    answer = ""

    while n > 0:
        n -= 1
        answer = num[n % 3] + answer
        n //= 3

    return answer

File: JavaScript/programmers_038.js
function solution(left, right) {
    var answer = 0
    for(let i = left ; i <= right; i++) {
        var count = 0
        for(let j = 1; j <= i; j++) {
            if(i % j === 0) {
                count++;
            }
        }
        if(count % 2 == 0) {
            answer += i
        }else if(count % 2 != 0) {
            answer -= i
        }
    }
    return answer;
}

// 제곱근이 정수로 떨어진다면 약수는 홀수

function solution(left, right) {
    var answer = 0;
    for (let i = left; i <= right; i++) {
        if (Number.isInteger(Math.sqrt(i))) {
            answer -= i;
        } else {
            answer += i;
        }
    }
    return answer;
}